simde/arm/neon/st1_x2.h

```diff
@@ -0,0 +1,190 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Copyright:
+ *   2020      Evan Nemerson <evan@nemerson.com>
+ *   2021      Décio Luiz Gazzoni Filho <decio@decpp.net>
+ */
+
+#if !defined(SIMDE_ARM_NEON_ST1_X2_H)
+#define SIMDE_ARM_NEON_ST1_X2_H
+
+#include "types.h"
+
+HEDLEY_DIAGNOSTIC_PUSH
+SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
+SIMDE_BEGIN_DECLS_
+
+SIMDE_FUNCTION_ATTRIBUTES
+void
+simde_vst1_f32_x2(simde_float32 ptr[HEDLEY_ARRAY_PARAM(4)], simde_float32x2x2_t val) {
```

Comment:
icc complained about this: `identifier "simde_float32x2x2_t" is undefined`

https://github.com/simd-everywhere/simde/pull/922/checks?check_run_id=3732776539#step:8:928

Reply:
I reviewed the relevant code in [simde/types/arm/neon.h](https://github.com/dgazzoni/simde/blob/st1_x234/simde/arm/neon/types.h). It turns out that the `...x{2,3,4}_t` structs are defined in either one of two places: on the block starting [here](https://github.com/dgazzoni/simde/blob/972c99c433b9bfb7afd58facdc491cf4fb038ca6/simde/arm/neon/types.h#L319) or the block starting [here] (https://github.com/dgazzoni/simde/blob/972c99c433b9bfb7afd58facdc491cf4fb038ca6/simde/arm/neon/types.h#L658). The first one is conditional on `SIMDE_ARM_NEON_A32V7_NATIVE`, which certainly isn't the case on icc. As for the second one, it is conditional on `SIMDE_BUG_INTEL_857088`. I assume the bug in question is present in the tested version of icc, so this is why neither block is used, and so the definition can't be found.

Following the pattern found in other files that also use `...x{2,3,4}_t` structs, I made all functions conditional on `SIMDE_BUG_INTEL_857088`, so they won't get compiled under icc.